app/models/knight.rb

```diff
@@ -1,4 +1,54 @@
 # frozen_string_literal: true
 
 class Knight < Piece
+# DETERMINE IF MOVE IS VALID
+# KNIGHT CAN MOVE TO 8 PLACES
+# EACH PLACE A HORIZONTAL OR VERTICLE L FROM CURRENT POSITION
+# CHECK IF OBSTRUCTED
+# OBSTRUCTED IF NEW POSITION HAVE PIECE OF SAME COLOR ON IT
+
+def valid_move? (move_to_x, move_to_y)
+
+  if (self.x_position - move_to_x) === 1 && (self.y_position - move_to_y) === -2
+    return true
+  elsif (self.x_position - move_to_x) === 2 && (self.y_position - move_to_y) === -1
+    return true
+  elsif (self.x_position - move_to_x) === 2 && (self.y_position - move_to_y) === 1
+    return true
+  elsif (self.x_position - move_to_x) === 1 && (self.y_position - move_to_y) === 2
+    return true
+  elsif (self.x_position - move_to_x) === -1 && (self.y_position - move_to_y) === 2
+    return true
+  elsif (self.x_position - move_to_x) === 2 && (self.y_position - move_to_y) === 1
+    return true
+  elsif (self.x_position - move_to_x) === -2 && (self.y_position - move_to_y) === -1
+    return true
+  elsif (self.x_position - move_to_x) === -1 && (self.y_position - move_to_y) === -2
+    return true

```

Comment:
Farouk,  I think lines 12 through 26 can be condensed down if you use the built in rails ".abs" method like you have in the king model.

Reply:
great suggestion - thanks!  have made the changes - much simpler - and will resubmit.